jetty/src/main/java/com/linecorp/armeria/server/jetty/ArmeriaConnector.java

```diff
@@ -16,65 +16,42 @@
 
 package com.linecorp.armeria.server.jetty;
 
-import java.util.Collection;
+import java.io.IOException;
+import java.nio.channels.ServerSocketChannel;
 import java.util.Collections;
 import java.util.List;
-import java.util.concurrent.Executor;
 import java.util.concurrent.Future;
 
-import javax.annotation.Nullable;
-
-import org.eclipse.jetty.io.ArrayByteBufferPool;
-import org.eclipse.jetty.io.ByteBufferPool;
 import org.eclipse.jetty.io.Connection;
 import org.eclipse.jetty.io.EndPoint;
 import org.eclipse.jetty.server.ConnectionFactory;
 import org.eclipse.jetty.server.Connector;
 import org.eclipse.jetty.server.HttpConfiguration;
 import org.eclipse.jetty.server.Server;
-import org.eclipse.jetty.util.component.ContainerLifeCycle;
-import org.eclipse.jetty.util.thread.ScheduledExecutorScheduler;
-import org.eclipse.jetty.util.thread.Scheduler;
+import org.eclipse.jetty.server.ServerConnector;
 
 import io.netty.util.concurrent.GlobalEventExecutor;
 
-final class ArmeriaConnector extends ContainerLifeCycle implements Connector {
+final class ArmeriaConnector extends ServerConnector {
 
     private static final String PROTOCOL_NAME = "armeria";
     private static final List<String> PROTOCOL_NAMES = Collections.singletonList(PROTOCOL_NAME);
 
-    private final Server server;
+    private final com.linecorp.armeria.server.Server armeriaServer;
     private final HttpConfiguration httpConfig;
-    private final Executor executor;
-    private final Scheduler scheduler;
-    private final ByteBufferPool byteBufferPool;
-    private final ArmeriaConnectionFactory connectionFactory;
-    private final Collection<ConnectionFactory> connectionFactories;
 
     private volatile boolean isShutdown;
 
-    ArmeriaConnector(Server server) {
-        this.server = server;
-        executor = server.getThreadPool();
+    ArmeriaConnector(Server server, com.linecorp.armeria.server.Server armeriaServer) {
+        super(server, 0, 0, new ArmeriaConnectionFactory());
```

Comment:
I found that `-1` is default value for `acceptors`, and `selectors`.
If you are not intended, how about passing `-1` to super method arguments?
 
https://github.com/eclipse/jetty.project/blob/fcc18b05300f0e3a9e357da2b9509f2768fc7011/jetty-server/src/main/java/org/eclipse/jetty/server/ServerConnector.java#L94-L104

Reply:
IIRC 0 and -1 have different meaning, but it may not matter anymore since we override most methods.